packages/react-core/src/components/Modal/ModalContent.tsx

```diff
@@ -88,24 +106,24 @@ export const ModalContent: React.FunctionComponent<ModalContentProps> = ({
   const modalBody = hasNoBodyWrapper ? (
     children
   ) : (
-    <ModalBoxBody {...props} {...(!description && { id })}>
+    <ModalBoxBody {...props} {...(!description && !ariaDescribedby && { descriptorId })}>
```

Comment:
Change to `<ModalBoxBody {...props} {...(!description && !ariaDescribedby && { id: descriptorId })}>`.